### PR TITLE
[MERGE] merge facl branch to master

### PR DESCRIPTION
**Pull request by yuzhu**

### CODE REVIEW

**Comment by yuzhu** [comment]
@gpang merge facl branch to master, PTAL.

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
Jenkins, test this please

**Comment by yuzhu** [comment]
@gpang PTAL, conflict-free merge

**Comment by gpang** [comment]
I think @calvinjia should also take a look.

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeDirectory.java [hunk 1]
```diff
@@ -50,6 +53,12 @@ public Object getFieldValue(Inode<?> o) {
 
   private boolean mDirectChildrenLoaded;
 
+  /**
+   * Empty property is used to indicate whether the defaultACL is empty or not.
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeDirectory.java]: Is this comment accurate?

File: core/common/src/main/java/alluxio/security/authorization/AclEntryParser.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.security.authorization;
+
+/**
+ * Parser for {@link AclEntry}.
+ */
+public final class AclEntryParser {
```
[the remark this answers is not in the document]
Comment (yuzhu) [reply core/common/src/main/java/alluxio/security/authorization/AclEntryParser.java]: done.

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeDirectory.java (hunk 1 above)
Comment (yuzhu) [reply core/server/master/src/main/java/alluxio/master/file/meta/InodeDirectory.java]: I think so

**Pull request closed without merging**